AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -2664,6 +2668,67 @@ - (void)_enqueueAsyncSizingWithSentinel:(ASSentinel *)sentinel completion:(void(
   return self;
 }
 
+#pragma mark - ASEnvironment
+
+- (ASEnvironmentCollection *)environmentCollection
+{
+  return &_environmentCollection;
+}
+
+- (ASDisplayNode *)parent
+{
+  return self.supernode;
+}
+
+- (void)setParent:(ASDisplayNode *)parent
+{
+  [self __setSupernode:parent];
+}
+
+- (NSArray<ASDisplayNode *> *)children
+{
+  return self.subnodes;
+}
+
+- (BOOL)supportsMultipleChildren
+{
+  return NO;
+}
+
+
+#pragma mark - ASLayoutableExtensibility
+
+- (void)setLayoutOptionExtensionBool:(BOOL)value atIndex:(int)idx
```

Comment:
Should we implement at least these methods with a #define in the same way that the ASLayoutable property accessors are?  In fact, this could go in ASLayoutable.mm ... since ASStackLayoutable is handled the same way, and is comparable to ASLayoutableExtensibility.